File: Stack/stack.js
var Stack = function(){

    this.items = []
};

Stack.prototype.size = function(){
	return this.items.length
}
Stack.prototype.isEmpty = function(){
	return this.items.length == 0
}
Stack.prototype.push = function(item){
	this.items.push(item)
}
Stack.prototype.pop = function(item){
	return this.items.pop()
}
Stack.prototype.peek = function(){
	return this.items[this.items.length-1]
}
Stack.prototype.size = function(){
	return this.items.length
}



function revstring(word){
	s = new Stack();
	
	for(var letter in word){
		s.push(word[letter]);
	}
	var reverse =""	;
	
	while( !s.isEmpty() ){
		reverse += s.pop();
	}
	return reverse;
}

function parenthesisParser(symbolString){
	s = new Stack();
	var correspond = true;
	var index = 0;
	
	while( index < symbolString.length && correspond){
		var symbol = symbolString[index];
		
			
		if (symbol == "("){
			s.push(symbol);
		} else {
			if (s.isEmpty()){
				correspond = false;
			} else {
				s.pop();				
			}		
		}
		index++;
	}
	if (correspond && s.isEmpty() ){
		return true;
	} else {
		return false;
	}
	
}

function matches(open, close){
	var opens = "([{";
	var closes = ")]}";
	return opens.indexOf(open) == closes.indexOf(close)
}

function symbolParser(symbolString){
	s = new Stack();
	var correspond = true;
	var index = 0;
	
	while( index < symbolString.length && correspond){
		var symbol = symbolString[index];
		if ( "({[".indexOf(symbol) != -1 ){
			s.push(symbol);
		} else {
			if (s.isEmpty()){

				correspond = false;
			} else {
				var top = s.pop();				
				if (!matches(top, symbol)){

					correspond = false;
				}
			}		
		}
		index++;
	}
	if (correspond && s.isEmpty() ){
		return true;
	} else {
		return false;
	}
	
}

File: Stack/stack.py
class Stack:
    def __init__(self):
        self.items = []

    def isEmpty(self):
        return self.items == []

    def push(self, item):
        self.items.append(item)

    def pop(self):
        return self.items.pop()

    def peek(self):
        """
        :return: Shows last item in the stack
        """
        return self.items[len(self.items)-1]

    def size(self):
        return len(self.items)



def revstring(word):
    """return a reversed string using Stack"""
    s = Stack()
    for letter in word:
        s.push(letter)
    reverse = ""
    while not s.isEmpty():
        reverse += s.pop()
    return reverse


def parenthesisParser(symbolString):
    """parses a string and tries to see it open and closed parenthesis correspond to eachother"""
    s = Stack()
    correspond = True
    index = 0
    while index < len(symbolString) and correspond:
        symbol = symbolString[index]
        if symbol == "(":
            s.push(symbol)
        else:
            if s.isEmpty():
                correspond = False
            else:
                s.pop()
        index += 1
    if correspond and s.isEmpty():
        return True
    else:
        return False

def matches(open, close):
    """ Returns true if the open and close params correspond to their symbols '([{' opening and closing repreentations"""
    opens = "[{("
    closers = "]})"
    return opens.index(open) == closers.index(close) 


def symbolParser(symbolString):
    """ parses a string to see if the '(,[,{' symbols match their closing parts """

    s = Stack()
    correspond = True
    index = 0
    while index < len(symbolString) and correspond:
        symbol = symbolString[index]
        if symbol in "([{":
            s.push(symbol)
        else:
            if s.isEmpty():
                correspond = False
            else:
                top = s.pop()
                if not matches(top, symbol):
                    correspond = False
        index += 1
    if correspond and s.isEmpty():
        return True
    else:
        return False

